src/Arcus.Security.Core/ISecretStore.cs

```diff
@@ -51,12 +46,13 @@ public interface ISecretStore
         /// <param name="name">The name that was used to register the <see cref="ICachedSecretProvider"/> in the secret store.</param>
         /// <typeparam name="TCachedSecretProvider">The concrete <see cref="ICachedSecretProvider"/> type.</typeparam>
         /// <exception cref="ArgumentException">Thrown when the <paramref name="name"/> is blank.</exception>
-        /// <exception cref="KeyNotFoundException">
-        ///     Thrown when there was no <see cref="ICachedSecretProvider"/> found in the secret store with the given <paramref name="name"/>,
-        ///     or there were multiple <see cref="ICachedSecretProvider"/> instances registered with the same name.
+        /// <exception cref="KeyNotFoundException">Thrown when there was no <see cref="ICachedSecretProvider"/> found in the secret store with the given <paramref name="name"/>.</exception>
+        /// <exception cref="NotSupportedException">
+        ///     Thrown when their was either none of the registered secret providers are registered as <see cref="ICachedSecretProvider"/> instances
+        ///     or there was an <see cref="ISecretProvider"/> registered but not with caching.
         /// </exception>
-        /// <exception cref="NotSupportedException">Thrown when there was an <see cref="ICachedSecretProvider"/> registered but not with caching.</exception>
         /// <exception cref="InvalidCastException">Thrown when the registered <see cref="ICachedSecretProvider"/> cannot be cast to the specific <typeparamref name="TCachedSecretProvider"/>.</exception>
+        /// <exception cref="InvalidOperationException">Thrown when multiple <see cref="ICachedSecretProvider"/> were registered with the same name.</exception>
```

Comment:
1. It's thrown (as explained above), because when accessing a specific concrete secret provider, we will have to cast it, and multiple registrations will result in a composite, so we can't cast it.
2. Yes, of course, you can register multiple cached secret providers with the same name too, and retrieve them with `.GetCachedSecretProvider`. But if you want an specific instance, you'll have to specify it with a unique name.

Reply:
Why would I use `GetCachedSecretProvider` instead of `GetSecretProvider<ICachedSecretProvider>` ?  I mean, what's the added value of this method ?